tests/workloads/test_dataframe.py

```diff
@@ -0,0 +1,60 @@
+from dask.sizeof import sizeof
+from dask.utils import format_bytes
+
+from ..utils_test import cluster_memory, timeseries_of_size, wait
+
+
+def print_dataframe_info(df):
+    p = df.partitions[0].compute(scheduler="threads")
+    partition_size = sizeof(p)
+    total_size = partition_size * df.npartitions
+    print(
+        f"~{len(p) * df.npartitions:,} rows x {len(df.columns)} columns, "
+        f"{format_bytes(total_size)} total, "
+        f"{df.npartitions:,} {format_bytes(partition_size)} partitions"
+    )
+
+
+def test_dataframe_align(small_client):
+    memory = cluster_memory(small_client)  # 76.66 GiB
+
+    df = timeseries_of_size(
+        memory // 2,
+        start="2020-01-01",
+        freq="600ms",
+        partition_freq="12h",
+        dtypes={i: float for i in range(100)},
+    )
+    print_dataframe_info(df)
+    # ~50,904,000 rows x 100 columns, 38.31 GiB total, 707 55.48 MiB partitions
+
+    df2 = timeseries_of_size(
+        memory // 4,
+        start="2010-01-01",
+        freq="600ms",
+        partition_freq="12h",
+        dtypes={i: float for i in range(100)},
+    )
+    print_dataframe_info(df2)
+    # ~25,488,000 rows x 100 columns, 19.18 GiB total, 354 55.48 MiB partitions
+
+    final = (df2 - df).mean()  # will be all NaN, just forcing alignment
+    wait(final, small_client, 10 * 60)
+
+
+def test_shuffle(small_client):
+    memory = cluster_memory(small_client)  # 76.66 GiB
+
+    df = timeseries_of_size(
+        memory // 4,
+        start="2020-01-01",
+        freq="1200ms",
+        partition_freq="24h",
+        dtypes={i: float for i in range(100)},
+    )
+    print_dataframe_info(df)
```

Comment:
How expensive do these local computes get? Is it ever a non-negligible amount of the test runtime?

Reply:
I think it's negligible (haven't actually timed it, but never notice it locally). And since the partition size is fixed, it should always remain negligible.

Nonetheless, I'm a little iffy on using the automatic timing from `small_client`; I might kinda prefer using `benchmark_time` manually. But as we start adding more things to benchmark, the Cartesian explosion of `small_client_time`, `small_client_time_memory`, etc. gets massive. Maybe we just have a `small_client` and `small_client_no_benchmarks` fixture, and you have to add every fixture manually in the latter case? That's annoying though, especially when adding a new thing to measure later.

Reply:
Sure, I'm happy with using the `benchmark_time` fixture manually. I liked having the "auto" fixture for ease of use to get off the ground quickly, but I see no problem with more custom stuff per test. I also wouldn't worry too much about re-using the `small_client` fixture. It wouldn't be hard to create a module-scoped cluster fixture that does your own thing for these workflows.

Is your worry only about the "wall clock" timing measuring some extra local work + client/cluster communication? If so, I might argue that that measure will always be a bit noisy, and it would be worth it to push on some [task group timing statistics](https://gist.github.com/ian-r-rose/b826ef75e3cdc6e83e3b31878eaf1305) as a more cluster-centric view on timing data. Those two measures would be pretty complementary.

Reply:
> Is your worry only about the "wall clock" timing measuring some extra local work + client/cluster communication?

Yeah, it is, which I agree isn't a big deal and should fall within the normal noise of runtime anyway.